### PR TITLE
Remove parameter objects

### PR DESCRIPTION
**Pull request by smitpatel**
Remove ReverseEngineerConfiguration
Remove CustomConfiguration

### CODE REVIEW

File: src/EFCore.Design/Design/Internal/DatabaseOperations.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ public class DatabaseOperations
             _servicesBuilder = new DesignTimeServicesBuilder(startupAssembly);
         }
 
-        public virtual Task<ReverseEngineerFiles> ScaffoldContextAsync(
+        public virtual Task<IEnumerable<string>> ScaffoldContextAsync(
```
Comment (bricelam) [inline src/EFCore.Design/Design/Internal/DatabaseOperations.cs]: Keep this one. I want to add stronger semantics to it. (i.e. ContextFile and EntityTypeFiles)

Comment (bricelam) [inline src/EFCore.Design/Design/Internal/DatabaseOperations.cs]: See #7263

**Comment by smitpatel** [comment]
@bricelam - Updated.

File: test/EFCore.Design.Tests/Scaffolding/Internal/ReverseEngineeringGeneratorTests.cs [hunk 2]
```diff
@@ -15,11 +15,10 @@ public class ReverseEngineeringGeneratorTests
...
         {
-            var outputNamespaceAndPaths = ReverseEngineeringGenerator.ConstructNamespaceAndCanonicalizedPaths(
-                rootNamespace, projectPath, outputPath);
-            Assert.Equal(expectedNamepace, outputNamespaceAndPaths.Namespace);
-            Assert.Equal(canonicalizedFullOutputPath, outputNamespaceAndPaths.CanonicalizedFullOutputPath);
-            Assert.Equal(canonicalizedRelativeOutputPath, outputNamespaceAndPaths.CanonicalizedRelativeOutputPath);
+            //var outputNamespaceAndPaths = ReverseEngineeringGenerator.ConstructNamespaceAndCanonicalizedPaths(
+            //    rootNamespace, projectPath, outputPath);
+            //Assert.Equal(expectedNamepace, outputNamespaceAndPaths.Namespace);
+            //Assert.Equal(canonicalizedFullOutputPath, outputNamespaceAndPaths.CanonicalizedFullOutputPath);
```
Comment (bricelam) [inline test/EFCore.Design.Tests/Scaffolding/Internal/ReverseEngineeringGeneratorTests.cs]: TODO?

Comment (smitpatel) [inline test/EFCore.Design.Tests/Scaffolding/Internal/ReverseEngineeringGeneratorTests.cs]: Ah. Thanks for reminding this bit. 
What should we do with these tests? We have removed function this test was testing.

Comment (bricelam) [inline test/EFCore.Design.Tests/Scaffolding/Internal/ReverseEngineeringGeneratorTests.cs]: Remove the test

**Pull request merged**